connectors/sources/sharepoint.py

```diff
@@ -666,71 +484,357 @@ async def get_drive_items(self, list_id, site_url, server_relative_url):
+                "order": 4,
+                "type": "str",
+                "value": "",
+            },
+            "client_secret": {
+                "depends_on": [{"field": "data_source", "value": SHAREPOINT_ONLINE}],
+                "label": "SharePoint Online secret id",
+                "order": 5,
+                "type": "str",
+                "value": "",
+            },
+            "tenant": {
+                "depends_on": [{"field": "data_source", "value": SHAREPOINT_ONLINE}],
+                "label": "SharePoint Online tenant",
+                "order": 6,
+                "type": "str",
+                "value": "",
+            },
+            "tenant_id": {
+                "depends_on": [{"field": "data_source", "value": SHAREPOINT_ONLINE}],
+                "label": "SharePoint Online tenant id",
+                "order": 7,
+                "type": "str",
+                "value": "",
+            },
+            "host_url": {
+                "label": "SharePoint host url",
+                "order": 8,
+                "type": "str",
+                "value": "http://127.0.0.1:8491",
+            },
+            "site_collections": {
+                "display": "textarea",
+                "label": "Comma-separated list of SharePoint site collections to index",
+                "order": 9,
+                "type": "list",
+                "value": "collection1",
+            },
+            "ssl_enabled": {
+                "display": "toggle",
+                "label": "Enable SSL",
+                "order": 10,
+                "type": "bool",
+                "value": False,
+            },
+            "ssl_ca": {
+                "depends_on": [{"field": "ssl_enabled", "value": True}],
+                "label": "SSL certificate",
+                "order": 11,
+                "type": "str",
+                "value": "",
+            },
+            "retry_count": {
+                "default_value": RETRIES,
+                "display": "numeric",
+                "label": "Retries per request",
+                "order": 12,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": RETRIES,
+            },
+        }
+
+    async def close(self):
+        """Closes unclosed client session"""
+        await self.sharepoint_client.close_session()
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+
+        Raises:
+            Exception: Configured keys can't be empty.
+        """
+        logger.info("Validating SharePoint Configuration")
+
+        connection_fields = (
+            [
+                "host_url",
+                "site_collections",
+                "client_id",
+                "client_secret",
+                "tenant",
+                "tenant_id",
+            ]
+            if self.sharepoint_client.is_cloud
```

Comment:
`validate_config` should validate anything from `self.sharepoint_client`

Reply:
we are validating all the required fields on the basis of `is_cloud` True/False.